src/masternodes/rpc_accounts.cpp

```diff
@@ -2025,6 +2031,231 @@ UniValue executesmartcontract(const JSONRPCRequest& request) {
     return NullUniValue;
 }
 
+UniValue futureswap(const JSONRPCRequest& request) {
+    auto pwallet = GetWallet(request);
+
+    RPCHelpMan{"futureswap",
+               "\nCreates and submits to the network a futures contract" +
+               HelpRequiringPassphrase(pwallet) + "\n",
+               {
+                       {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "Address to fund contract and receive resulting token"},
+                       {"amount", RPCArg::Type::STR, RPCArg::Optional::NO, "Amount to send in amount@token format"},
```

Comment:
Could we accept multiple non-DUSD amounts ?

Reply:
Correct. You can swap dTokens for DUSD.

Reply:
I was not clear, I meant accepting multiple amounts at the same time. As in changing `CDFIP2203Message`: `CBalances amounts;` instead of `CTokenAmount source{};` to use a single TX

Reply:
Got it, the TX allows creating a contract for a single source token only.

Reply:
The rationale was because for DUSD source you need to set the destination, with dToken source this is not an issue. Wanted to keep a consistent interface.